### PR TITLE
APIMF-2679: parse message header into new header object field with no…

### PR DESCRIPTION
**Pull request by pope1838**
…de shape instead of build a virtual parameter

### CODE REVIEW

File: amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/parser/spec/async/parser/AsyncMessageParser.scala [hunk 1]
```diff
@@ -149,9 +150,18 @@ abstract class AsyncMessagePopulator()(implicit ctx: AsyncWebApiContext) extends
...
-        parseHeaderSchema(entry, message.id) foreach { param =>
-          message.set(MessageModel.Headers, AmfArray(Seq(param), Annotations(entry.value)), Annotations(entry))
+      entry => {
+        AsyncApiTypeParser(entry, shape => shape.withName("schema").adopted(message.id), JSONSchemaDraft7SchemaVersion)
+          .parse()
+          .foreach {
+            case n: NodeShape =>
+              message.set(MessageModel.HeaderSchema, n, Annotations(entry))
+            case _ =>
+              ctx.eh.violation(ParserSideValidations.HeaderMustBeObject,
```
Comment (tomsfernandez) [inline amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/parser/spec/async/parser/AsyncMessageParser.scala:162]: [OOF] - A good improvement is to stop parsing the shape once we know its a nodeShape as we don't assign it the instance if it isn't a NodeShape. Probably not significant but still a processing saving

File: amf-webapi/shared/src/main/scala/amf/plugins/domain/webapi/models/Message.scala [hunk 2]
```diff
@@ -28,6 +28,7 @@ class Message(override val fields: Fields, override val annotations: Annotations
   def bindings: MessageBindings    = fields.field(Bindings)
   def headerExamples: Seq[Example] = fields.field(HeaderExamples)
   def headers: Seq[Parameter]      = fields.field(Headers)
+  def headerObject: NodeShape      = fields.field(HeaderSchema)
```
Comment (tomsfernandez) [inline amf-webapi/shared/src/main/scala/amf/plugins/domain/webapi/models/Message.scala]: headerObject should also be renamed to HeaderSchema

**Review by tomsfernandez — CHANGES_REQUESTED** [review]
Please also update client side interface.

[OOF] - Update typings jeje

File: amf-webapi/shared/src/main/scala/amf/client/model/domain/Message.scala [hunk 3]
```diff
@@ -30,6 +30,7 @@ class Message(override private[amf] val _internal: InternalMessage)
   def title: StrField                     = _internal.title
   def summary: StrField                   = _internal.summary
   def bindings: MessageBindings           = _internal.bindings
+  def headerSchema: NodeShape             = _internal.headerSchema
```
Comment (AgustinBettati) [inline amf-webapi/shared/src/main/scala/amf/client/model/domain/Message.scala:33]: missing withHeaderSchema method in client interface!

**Review by AgustinBettati — CHANGES_REQUESTED** [review]

**Review by tomsfernandez — APPROVED** [review]

**Review by AgustinBettati — APPROVED** [review]

**Pull request merged**